crates/svm-runtime/src/runtime/default.rs

```diff
@@ -355,21 +355,16 @@ where
         }
     }
 
-    fn init_instance_buffer(&self, func_buf: &Vec<u8>, instance: &mut wasmer_runtime::Instance) {
-        const ARGS_BUF_ID: u32 = 0;
-
+    fn copy_func_buf_to_memory(&self, func_buf: &[u8], instance: &mut wasmer_runtime::Instance) {
         let ctx = instance.context_mut();
+        let memory = ctx.memory(0);
 
-        helpers::buffer_create(ctx.data, ARGS_BUF_ID, func_buf.len() as u32);
+        let func_size = func_buf.len();
+        let view = &memory.view::<u8>()[0..func_size];
```

Comment:
What about bound checking for the memory size?

Reply:
this should not happen here.
when we allocate memory we know the size of the `func_buf`

Reply:
But you're currently don't check it there - do you mean that you will add that check later, or that you'll verify that `func_buf` size is less than 1 page size?

Reply:
* The `validate_template` will check that `func_buf` isn't big enough.
* The `alloc_wasmer_memory` will have to allocate enough memory for the program to run.

I've added an issue: #140